sdcm/nemesis.py

```diff
@@ -4568,9 +4568,20 @@ def disrupt_create_index(self):
             if not ks_cf_list:
                 raise UnsupportedNemesis("No table found to create index on")
             ks, cf = random.choice(ks_cf_list).split('.')
-            column = get_random_column_name(session, ks, cf)
-            if not column:
+            all_column_names = get_column_names(session, ks, cf)
+
+            # Query the system_schema.indexes table to get a list of columns with secondary indexes
+            index_query = f"SELECT column_name FROM system_schema.indexes WHERE keyspace_name = '{ks}' AND table_name = '{cf}'"
+            index_res = session.execute(index_query)
+            columns_with_secondary_index = {row.column_name for row in index_res}
+
+            # Filter out columns with secondary indexes
+            filtered_column_names = [column_name for column_name in all_column_names if
```

Comment:
Index can not be created on partition key. 
To find partition key column you can use `sdcm.utils.common.get_partition_keys`

Reply:
Also the module `sdcm.utils.nemesis_utils.indexes.get_column_names` can be usefull